src/main/scala/beam/agentsim/infrastructure/SplitParkingManager.scala

```diff
@@ -0,0 +1,269 @@
+import beam.agentsim.infrastructure.taz.{TAZ, TAZTreeMap}
+import beam.sim.common.GeoUtils
+import beam.sim.common.GeoUtils.toJtsCoordinate
+import beam.sim.config.BeamConfig
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.algorithm.ConvexHull
+import com.vividsolutions.jts.geom.{Coordinate, Envelope, Geometry, GeometryFactory}
+import de.lmu.ifi.dbs.elki.algorithm.clustering.kmeans.KMeansElkan
+import de.lmu.ifi.dbs.elki.algorithm.clustering.kmeans.initialization.RandomUniformGeneratedInitialMeans
+import de.lmu.ifi.dbs.elki.data.`type`.TypeUtil
+import de.lmu.ifi.dbs.elki.data.{DoubleVector, NumberVector}
+import de.lmu.ifi.dbs.elki.database.ids.{DBIDIter, DBIDRef}
+import de.lmu.ifi.dbs.elki.database.relation.Relation
+import de.lmu.ifi.dbs.elki.database.{Database, StaticArrayDatabase}
+import de.lmu.ifi.dbs.elki.datasource.ArrayAdapterDatabaseConnection
+import de.lmu.ifi.dbs.elki.distance.distancefunction.minkowski.SquaredEuclideanDistanceFunction
+import de.lmu.ifi.dbs.elki.utilities.random.RandomFactory
+import org.matsim.api.core.v01.{Coord, Id}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.ExecutionContext
+import scala.util.Random
+
+/**
+  *
+  * @author Dmitry Openkov
+  */
+class SplitParkingManager(
+  beamConfig: BeamConfig,
+  tazTreeMap: TAZTreeMap,
+  clusters: Vector[ParkingCluster],
+  zones: Array[ParkingZone],
+  searchTree: ZoneSearchTree[TAZ],
+  geo: GeoUtils,
+  rand: Random,
+  boundingBox: Envelope
+) extends beam.utils.CriticalActor
+    with ActorLogging {
+  private implicit val timeout: Timeout = Timeout(100, TimeUnit.SECONDS)
+  private implicit val executionContext: ExecutionContext = context.dispatcher
+
+  private val workers: Vector[Worker] =
+    clusters.zipWithIndex.map {
+      case (cluster, i) =>
+        val actor = context.actorOf(
+          ZonalParkingManager
+            .props(
+              beamConfig,
+              TAZTreeMap.fromSeq(cluster.tazes),
+              zones,
+              searchTree,
+              geo,
+              rand,
+              boundingBox
+            ),
+          s"zonal-parking-manager-$i"
+        )
+        Worker(actor, cluster)
+    }
+
+  private val tazToWorker: Map[Id[TAZ], Worker] = mapTazToWorker(workers)
+
+  override def receive = {
+    case inquiry: ParkingInquiry =>
+      val foundCluster = workers.find { w =>
+        val point = SplitParkingManager.geometryFactory.createPoint(inquiry.destinationUtm)
+        w.cluster.convexHull.contains(point)
+      }
+
+      val worker =
+        foundCluster.getOrElse(
+          tazToWorker.getOrElse(
+            tazTreeMap.getTAZ(inquiry.destinationUtm.getX, inquiry.destinationUtm.getY).tazId,
+            throw new IllegalStateException(s"No TAZ around for $inquiry")
+          )
+        )
+
+      worker.actor.forward(inquiry)
```

Comment:
emergency stall: does it work properly in terms of accounting with multiple actors?